traffic_portal/app/src/common/modules/dialog/select/lock/dialog.select.lock.tpl.html

```diff
@@ -0,0 +1,69 @@
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+<div class="modal-header">
+    <button type="button" class="close" ng-click="cancel()"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
+    <h4 class="modal-title">Create CDN Lock</h4>
+</div>
+<div class="modal-body">
+    <form name="lockForm" novalidate>
+        <div class="row">
+            <div class="col-sm-12 col-md-12">
+                <div class="form-group" ng-class="{'has-error': hasError(lockForm.cdn), 'has-feedback': hasError(lockForm.cdn)}">
+                    <label class="control-label" for="cdn">CDN *
+                        <small class="input-error" ng-show="hasPropertyError(lockForm.cdn, 'required')">Required</small>
+                    </label>
+                    <select id="cdn" name="cdn" class="form-control" ng-model="lock.cdn" ng-options="cdn.name as cdn.name for cdn in cdns" required>
+                        <option value="">Select...</option>
+                    </select>
+                </div>
+            </div>
+        </div>
+        <div class="row">
+            <div class="col-sm-12 col-md-12">
+                <div class="form-group" ng-class="{'has-error': hasError(lockForm.message), 'has-feedback': hasError(lockForm.message)}">
+                    <label class="control-label" for="message">Message *
+                        <small class="input-error" ng-show="hasPropertyError(lockForm.message, 'maxlength')">Too Long</small>
+                    </label>
+                    <input id="message" name="message" type="text" class="form-control" ng-model="lock.message" ng-maxlength="256" required autofocus>
+                    <span ng-show="hasError(lockForm.message)" class="form-control-feedback"><i class="fa fa-times"></i></span>
+                </div>
+            </div>
+        </div>
+        <div class="row">
+                <div class="col-sm-12 col-md-12">
+                    <div class="form-group">
+                        <p>Type *</p>
+                        <input id="soft" type="radio" data-ng-model="lock.soft" data-ng-value="true">
+                        <label class="has-tooltip" for="soft">&nbsp;Soft<div class="helptooltip">
+                            <div class="helptext">A soft lock will prevent others from queueing server updates or snapshotting the locked CDN.</div>
+                        </div>
+                        </label><br>
+                        <input id="hard" type="radio" data-ng-model="lock.soft" data-ng-value="false">
```

Comment:
Does this work with the `data-` prefixes? At the least it's unnecessary. Also, in order to make the browser recognize these as mutually exclusive options, they should have the same `name` attribute. Finally, these should have the `required` attribute to make it clear that a selection must be made (it's best to put it on both, IMO, for clarity, but when they have the same `name` only one _needs_ to have it for it to work strictly speaking)

Reply:
i added the data- prefixes because the google told me to because it wasn't binding to ng-model or something like that. tbh, can't remember the last time i used a radio button. you want to take a stab at writing that radio buttons block?

Reply:
seems you got it to work without `data-`